### PR TITLE
add support for manually updating subscription statuses

### PR DESCRIPTION
**Pull request by hankim813**
Now you should be able to update a given user's subscription status by including an exclusive `Mailchimp` object in the `options` parameter of `.identify()` calls, just like how you could pass a custom `listId`:

```
analytics.identify('12345', {
  firstName: 'Peter'
  }, {
    "MailChimp": {
      "listId": "ea7918abb6",
      "subscriptionStatus": "unsubscribed"
    }
  });
```

IMPORTANT: this will be documented but you **must** use the exact property (case sensitive), `subscriptionStatus` and the value of this property must be one of the four valid statuses from MC, `pending`, `subscribed`, `unsubscribed`, `cleaned`.

Also -- this will only work for existing users. All new email addresses will either be set to `pending` or `subscribed` based on your `doubleOptIn` setting.

If you provide an invalid status, we will reject it and fallback on whatever the MC API returned for that user's current subscription status.

link to corresponding `site-docs` PR: https://github.com/segmentio/site-docs/pull/826

@f2prateek @sperand-io

### CODE REVIEW

**Comment by pauloschilling** [comment]
LGTM :+1:

Thank you again for this PR! :)

File: test/index.js [hunk 1]
```diff
@@ -182,6 +212,7 @@ describe('MailChimp', function(){
         .end(done);
     });
 
+
```
Comment (f2prateek) [inline test/index.js]: unecessary whitespace

File: lib/mapper.js [hunk 2]
```diff
@@ -31,6 +31,14 @@ exports.identify = function(identify){
     }
   };
 
+  // check if there is a custom subscription status provided
+  if (opts.subscriptionStatus) {
+    var validStatuses = ['pending', 'subscribed', 'unsubscribed', 'cleaned'];
```
Comment (f2prateek) [inline lib/mapper.js]: Do you think we need to check this? Users will always read this from the documentation and pass it in specifc to Mailchimp, if they send something else they made a mistake and we shouldn't be correcting it.

**Comment by f2prateek** [comment]
LGTM. Nothing blocking this change from going out, feel free to merge!

File: lib/mapper.js (hunk 2 above)
Comment (hankim813) [inline lib/mapper.js]: I think we can't hurt. I think it's cuz Mailchimp might throw away the `.identify()` call if the status is invalid.

**Comment by hankim813** [comment]
@pauloschilling we just deployed this! give it a try and let us know if something isn't working :)

here's the ref to our updated docs: https://segment.com/docs/integrations/mailchimp/#manually-updating-user-subscription-status

cc @sperand-io

**Comment by pauloschilling** [comment]
Hey @hankim813, @sperand-io, just passing by to say that I did a lot of testing using our staging environment and everything is working fine! Thank you guys!

**Pull request merged**